lib/bolt_spec/plans/action_stubs.rb

```diff
@@ -127,7 +153,10 @@ def return(&block)
       def return_for_targets(data)
         data.each_with_object(@data) do |(target, result), hsh|
           raise "Mocked results must be hashes: #{target}: #{result}" unless result.is_a? Hash
-          hsh[target] = result_for(Bolt::Target.new(target), Bolt::Util.walk_keys(result, &:to_sym))
+          # set the inventory from the BoltSpec::Plans, otherwise if we try to convert
+          # this target to a string, it will fail to string conversion because the
+          # inventory is nil
+          hsh[target] = result_for(Bolt::Target.new(target, @inventory), Bolt::Util.walk_keys(result, &:to_sym))
```

Comment:
Does the plan stub of `result_for` use the target parameter? I'm not sure in what cases this breaks if the inventory is nil...

Reply:
This would also mean we don't have to pass inventory here.

Reply:
no, it doesn't use the target parameter, it just silently ignores it: https://github.com/puppetlabs/bolt/pull/1631/files#diff-075d987aae32893fba7412f7e47a1b2dR33

Reply:
@lucywyman do you want me to change something here?

Reply:
We need to pass the inventory here, because otherwise if there is an error and this target is printed out, it will fail to convert the `target` to a string and throw a different exception (PAL Error).